sys/include/net/nanocoap.h

```diff
@@ -1790,6 +1790,22 @@ static inline size_t coap_opt_put_block2_control(uint8_t *buf, uint16_t lastonum
                              (block->blknum << 4) | block->szx);
 }
 
+/**
+ * @brief   Insert an CoAP Observe Option into the buffer
+ *
+ * @param[out]  buf         Buffer to write to
+ * @param[in]   lastonum    last option number (must be < 6)
+ * @param[in]   obs         observe number to write
+ *
+ * @param[in]   lastonum    last option number (must be < 27)
```

Comment:
must lastonum be smaller than 27 or 6?

Reply:
Good catch, fixed.

Reply:
What about the assert?

Reply:
There `assert()` is in `coap_put_option()` that gets called under the hood of `coap_put_opt_uint()`.